core/components/organisms/select/Select.tsx

```diff
@@ -211,11 +219,17 @@ export const Select = React.forwardRef<SelectMethods, SelectProps>((props, ref)
   }, [value]);
 
   const onToggleHandler = (open: boolean) => {
+    if (onToggle) {
+      onToggle(open);
+    }
+
     if (triggerOptions && triggerOptions.disabled) {
       setOpenPopover(false);
     } else {
-      setHighlightFirstItem(open);
       setOpenPopover(open);
+      setTimeout(() => {
+        setHighlightFirstItem(open);
+      }, 200);
```

Comment:
If `setHighlightFirstItem(open);` is dependent on the state of `open`, then please use `useEffect` hook instead of using setTimeout. 

```
React.useEffect(() => {
  setHighlightFirstItem(open);
}, [open])
```